runtime/caches.go

```diff
@@ -81,7 +84,12 @@ func (c *connectionCache) get(ctx context.Context, instanceID, driver, dsn strin
 		if instanceID != "default" {
 			logger = c.logger.With(zap.String("instance_id", instanceID), zap.String("driver", driver))
 		}
-		conn, err := drivers.Open(driver, map[string]any{"dsn": dsn}, logger)
+		config := map[string]any{
+			"dsn":          dsn,
+			"activity":     c.activity,
+			"activityDims": activityDims,
```

Comment:
1. Similar to logging, could we add something lightweight like `c.activity.With(dims)` to avoid propagating the dims alongside?
2. Instead of passing it in the `config`, it would be cleaner to pass it along with the logger to `Open`. But let's do it later to avoid too many merge conflicts with this PR: https://github.com/rilldata/rill/pull/2875. Will add an issue for it.

Reply:
Added a lightweight `wrappedClient` to hold dimension